model/riscv_insts_zicfilp.sail

```diff
@@ -0,0 +1,42 @@
+/*=======================================================================================*/
+/*  This Sail RISC-V architecture model, comprising all files and                        */
+/*  directories except where otherwise noted is subject the BSD                          */
+/*  two-clause license in the LICENSE file.                                              */
+/*                                                                                       */
+/*  SPDX-License-Identifier: BSD-2-Clause                                                */
+/*=======================================================================================*/
+
+function make_lpad_exception() -> sync_exception =
+  struct { trap    = E_Software_Check(),
+           excinfo = Some(enc_software_check(SWC_LANDING_PAD_FAULT)),
+           ext     = None() }
+
+union clause instruction = LPAD : (lpl)
+
+// LPAD is encoded as AUIPC with rd as x0
+mapping clause encdec = LPAD(lpl)
+  <-> lpl @ 0b00000 @ 0b0010111
+      when currentlyEnabled(Ext_Zicfilp)
+
+function clause execute LPAD(lpl) = {
+  if is_lp_expected() then {
+    let unaligned_pc = get_arch_pc()[1 .. 0] != 0b00;
+    let label_mismatch = X(Regidx(zero_extend(0x7)))[31 .. 12] != lpl & lpl != zeros();
```

Comment:
Shouldnt the zero_extend be on the context of x[7] and not on 7.

Reply:
`regidx_bit_width` has a variable bit-width of 4 or 5 depending on whether the base ISA is E or not, so `0x7` might need extending to 5 bits.  The context doesn't need it since both sides are 20 bits.